src/snowflake/cli/api/project/definition_conversion.py

```diff
@@ -89,9 +109,14 @@ def convert_project_definition_to_v2(
             native_app_data.get("entities", {}),
         ),
         "mixins": snowpark_data.get("mixins", None),
-        "env": envs,
     }
+    if envs is not None:
+        data["env"] = envs
```

Comment:
`None` isn't a valid value for `env` according to the pydantic model. When saving the converted PDF to disk, `None` is omitted but when converting in-memory, we need to omit it here.